#135 refactor: split livewire component and view files to separate ed…

KarbovskyiV opened the pull request:
…it and create, and create basic component and view file

KarbovskyiV committed:
#135 refactor: split livewire component and view files to separate edit and create, and create basic component and view file

KarbovskyiV committed:
#135 refactor: add getFilteredKeysFromConfig method to FormTrait

KarbovskyiV committed:
#135 feat: add check and displaying error if same code was selected

Vitaliy-1 commented on app/Livewire/Encounter/EncounterComponent.php:
```diff
@@ -4,16 +4,307 @@
+use App\Classes\eHealth\Api\PatientApi;
+use App\Classes\eHealth\Api\ServiceRequestApi;
+use App\Livewire\Encounter\Forms\Api\EncounterRequestApi;
+use App\Models\Employee\Employee;
```
Лишній імпорт

Vitaliy-1 commented on app/Livewire/Encounter/EncounterComponent.php:
```diff
@@ -4,16 +4,307 @@
+    public function searchForReferralNumber(): void
+    {
+        $buildSearchRequest = EncounterRequestApi::buildGetServiceRequestList($this->form->referralNumber);
+        $requisitionData = ServiceRequestApi::searchForServiceRequestsByParams($buildSearchRequest);
```
Я підозрюю, що в цьому та інших подібних місцях методів із void return type, декларувати змінну не треба

Vitaliy-1 commented on app/Livewire/Encounter/EncounterComponent.php:
```diff
@@ -4,16 +4,307 @@
+            ->select(['id', 'uuid', 'first_name', 'last_name', 'second_name'])
+            ->where('id', $this->patientId)
+            ->first()
+            ?->toArray();
```
Якщо patient буде null, то нижче буде fatal error

Vitaliy-1 replied:
Якщо `last_name` буде `null`, то вам поверне такий масив чи ні?
```php
[
    'uuid' => ...,
    'first_name' => ...,
    'last_name' => ...,
    'second_name' => null // Чи Eloquent це значення прибере?
]
```

KarbovskyiV replied:
last_name не може бути null, в БД це обов'язкове поле

Vitaliy-1 commented on app/Livewire/Encounter/EncounterComponent.php:
```diff
@@ -4,16 +4,307 @@
+        $this->patientUuid = $patient['uuid'];
+        $this->firstName = $patient['first_name'];
+        $this->lastName = $patient['last_name'];
+        $this->secondName = $patient['second_name'] ?? null;
```
До речі, мабуть також можна через array destructuring 
```php
[
    'uuid' => $this->patientUuid,
    'first_name' => $this->firstName,
    'last_name' => $this->lastName,
    'second_name' => $this->secondName
] = $patient;
```
Хоча такий варіант читається в даному випадку не дуже

Vitaliy-1 commented on app/Livewire/Encounter/EncounterComponent.php:
```diff
@@ -4,16 +4,307 @@
+     *
+     * @return void
+     */
+    protected function setPatientData(): void
```
Чи він потрібен цей метод в `EncounterCreate`?

KarbovskyiV replied:
Так, так як це інформація про пацієнта для якого створюється енкаунтер

Vitaliy-1 commented on app/Livewire/Encounter/EncounterComponent.php:
```diff
@@ -4,16 +4,307 @@
+     *
+     * @return void
+     */
+    protected function adjustEpisodeTypes(): void
```
Я бачу, що у вас цей та декілька інших методів використовуються в обох child класах - create та edit. В таких випадках краще його й викликати в parent класі, в mount методі. Хіба ви не плануєте його використовувати в view child класі...

KarbovskyiV replied:
Маунти відрізняються, зробив окремий метод в батьківському класі

Vitaliy-1 commented on app/Livewire/Encounter/EncounterEdit.php:
```diff
@@ -0,0 +1,91 @@
+     * @return void
+     * @throws Throwable
+     */
+    public function save(): void
```
Одразу краще читається!

Vitaliy-1 reviewed (changes requested):
Дякую, так виглядає краще! Залишив пару коментарів

KarbovskyiV committed:
#135 refactor: add initializeComponent method for init same data,
move public methods to the top of the component,
improve person db query,
small fixes,
chore: add rules to pint for imports
